extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -177,27 +184,160 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 
 	public async getDatabaseAssessments(): Promise<ServerAssessement> {
 		const ownerUri = await azdata.connection.getUriForConnection(this.sourceConnectionId);
-		// stress test backend & dialog component
-		const assessmentResults = await this.migrationService.getAssessments(
-			ownerUri,
-			this._databaseAssessment
-		);
-		const dbAssessments = assessmentResults?.assessmentResult.databases.filter(d => !this.excludeDbs.includes(d.name)).map(d => {
-			return {
-				name: d.name,
-				issues: d.items.filter(i => i.appliesToMigrationTargetPlatform === MigrationTargetType.SQLMI) ?? []
-			};
-		});
+		this._assessmentApiResponse = (await this.migrationService.getAssessments(ownerUri, this._databaseAssessment))!;
 		this._assessmentResults = {
-			issues: assessmentResults?.assessmentResult.items?.filter(i => i.appliesToMigrationTargetPlatform === MigrationTargetType.SQLMI) ?? [],
-			databaseAssessments: dbAssessments! ?? []
+			issues: this._assessmentApiResponse.assessmentResult.items,
+			databaseAssessments: this._assessmentApiResponse.assessmentResult.databases.map(d => {
+				return {
+					name: d.name,
+					issues: d.items
+				};
+			})
 		};
-
+		this.generateAssessmentTelemetry();
 		return this._assessmentResults;
 	}
 
-	public findDatabaseAssessments(databaseName: string): mssql.SqlMigrationAssessmentResultItem[] | undefined {
-		return this._assessmentResults.databaseAssessments.find(databaseAsssessment => databaseAsssessment.name === databaseName)?.issues;
+	private async generateAssessmentTelemetry(): Promise<void> {
+		try {
+			this._assessmentResults.issues.forEach(i => {
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.ServerAssessmentIssues,
+					{
+						'sessionId': this._sessionId,
+						'ruleId': i.ruleId
+					},
+					{
+						'count': i.impactedObjects.length
+					}
+				);
+			});
+
+			const serverAssessmentErrorsMap: Map<number, number> = new Map();
+			this._assessmentApiResponse.assessmentResult.errors.forEach(e => {
+				if (serverAssessmentErrorsMap.has(e.errorId)) {
+					serverAssessmentErrorsMap.set(e.errorId, serverAssessmentErrorsMap.get(e.errorId)! + 1);
+				} else {
+					serverAssessmentErrorsMap.set(e.errorId, 1);
+				}
+			});
+
+			serverAssessmentErrorsMap.forEach((v, k) => {
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.ServerAssessmentError,
+					{
+						'sessionId': this._sessionId
+					},
+					{
+						'errorId': k,
```

Comment:
The errorId isn't really a measurement, this should probably be in the properties bucket